assembly/src/lib.rs

```diff
@@ -24,7 +24,7 @@ use procedures::Procedure;
 
 mod parsers;
 use parsers::{combine_blocks, parse_code_blocks};
-pub use parsers::{parse_module, ModuleAst};
+pub use parsers::{ModuleAst, ProcedureAst, ProgramAst};
```

Comment:
I am not sure we need to expose `ProcedureAst` and `ProgramAst` publicly here. We do need `ModuleAst` for library serialization and deserialization - but others could probably stay internal (at least for now).

Reply:
Moved back to private scope